### ISSUE TITLE
Interactable not receiving input from Grip Press or Menu Button on an MR Controller

### ISSUE THREAD

**Issue by killerantz** [microsoft/MixedRealityToolkit-Unity#4492]
When testing a controller grip button or the menu button, it does not trigger any state or events on the Interactable. Sometimes I use an Interactable with IsGlobal set to true, then set the Mixed Reality Input Action to Menu, so the menu button on the controller will trigger an action.

1. Place an Interactable on a gameObject.
2. Add a collider
3. Set the Input Action drop down to Menu or Grip Press.
4. Test in Unity with a MR Headset
5. Press the menu button

![Interactable Inspector Menu action](https://user-images.githubusercontent.com/22085553/57867392-ecf86b00-77b5-11e9-8947-9ddfd94917ac.png)

![Interactable Inspector Grip Press action](https://user-images.githubusercontent.com/22085553/57867449-0a2d3980-77b6-11e9-8e96-4bb8c0c7c6f2.png)

## Intended Behavior

MR Controller buttons other than the trigger (select) can send input into and Interactable.

## Your Setup (please complete the following information)

- Unity Version [2018.3.9f1]
- MRTK Version [v2.0]

## Target Platform (please complete the following information)
- WMR immersive

## Additional context

@julenka did a test with the Xbox Controller, it worked mapping buttons from the Xbox controller to an custom action, there may be an issue with existing actions?

**Comment by wiwei** [comment]
@julenka what's the impact of this issue?

**Comment by julenka** [comment]
So far I think it is just impacting Todd, since most people use interactable for just the select button. I also think it could just be that Todd is not configuring the interactable in a way that matches the way I would do it, since it was working for me when I checked it in. So I would say it is impacting one person but it’s possible that he is just configuring it incorrectly.

Sent from phone, please excuse typos.
________________________________

**Comment by killerantz** [comment]
I tracked it down to this method in BaseControllerPointer.

```
public override void OnInputDown(InputEventData eventData)
        {
            base.OnInputDown(eventData);

            if (eventData.SourceId == InputSourceParent.SourceId)
            {
                if (requiresHoldAction && eventData.MixedRealityInputAction == activeHoldAction)
                {
                    IsHoldPressed = true;
                }

                if (eventData.MixedRealityInputAction == pointerAction)
                {
                    IsSelectPressed = true;
                    HasSelectPressedOnce = true;

                    if (IsInteractionEnabled)
                    {
                        InputSystem.RaisePointerDown(this, pointerAction, Handedness);
                    }
                }
            }
        }
```

The check for (eventData.MixedRealityInputAction == pointerAction) is false so the Menu button will not raise any pointer events.

Called on  |  Action Description  |  Axis Constraint
------------|-----------------------|------------------
Right_DefaultControllerPointer(Clone)  |  Select |  Digital
Right_ParabolicPointer(Clone) |  Trigger  |  SingleAxis
Menu Action Button  |  Menu  |  Digital

I don't know much about this or if there is something I should be setting up or registering?

**Comment by julenka** [comment]
Oh, okay I understand the bug now. Let me think about it. Thanks for digging into this!

**Comment by julenka** [comment]
Understanding Todd's comment a bit better, this means that it's not possible to listen to any interactions other than 'select' currently, which I do think could impact a large number of customers (anybody listening to an action that is not select), and is a regression. Adding Urgency-Soon tag and taking a look.

**Comment by julenka** [comment]
I figured out why this was working for the gamepad and not for motion controllers. Here's the code for GGVPointer, which is what the gamepad uses:

```
        public void OnInputDown(InputEventData eventData)
        {
            for (int i = 0; i < eventData.InputSource.Pointers.Length; i++)
            {
                if (eventData.InputSource.Pointers[i].PointerId == GazePointer.PointerId)
                {
                    gazePointer.RaisePointerDown(eventData.MixedRealityInputAction, eventData.Handedness, eventData.InputSource);
                    return;
                }
            }
        }
```

They are not filtering based on the 'select action' type

**Comment by julenka** [comment]
Fixed by #4673
